observability-lib/README.md

```diff
@@ -1,8 +1,118 @@
-## Observability-lib
+# Observability-lib
 
-Contains the observability library to build dashboards and alerts
+This library enables creating Grafana dashboards and alerts with go code.
+
+It provides abstractions to create grafana resources :
+- [Dashboards](https://grafana.com/docs/grafana/latest/dashboards/)
+- [Alerts](https://grafana.com/docs/grafana/latest/alerting/)
+- [Contact Points](https://grafana.com/docs/grafana/latest/alerting/fundamentals/notifications/contact-points/)
+- [Notification Policies](https://grafana.com/docs/grafana/latest/alerting/configure-notifications/create-notification-policy/)
+- [Notification Templates](https://grafana.com/docs/grafana/latest/alerting/configure-notifications/template-notifications/create-notification-templates/)
+
+## Folder Structure
+
+The observability-lib is structured as follows:
+```shell
+observability-lib/
+    api/ # Grafana HTTP API Client to interact with resources
+    cmd/ # CLI to interact deploy or generateJSON from dashboards defined in folder below
+    dashboards/ # Dashboards definitions
+    grafana/ # grafana-foundations-sdk abstraction to manipulate grafana resources
+```
+
+## Quickstart
+
+### Creating a dashboard
+
+```go
+package main
+
+import "github.com/smartcontractkit/chainlink-common/observability-lib/grafana"
+
+builder := grafana.NewBuilder(&grafana.BuilderOptions{
+    Name:       "Dashboard Name",
+    Tags:       []string{"tags1", "tags2"},
+    Refresh:    "30s",
+    TimeFrom:   "now-30m",
+    TimeTo:     "now",
+})
+
+builder.AddVars(grafana.NewQueryVariable(&grafana.QueryVariableOptions{
+    VariableOption: &grafana.VariableOption{
+        Label: "Environment",
+        Name:  "env",
+    },
+    Datasource: "Prometheus",
+    Query:      `label_values(up, env)`,
+}))
+
+builder.AddRow("Summary")
+
+builder.AddPanel(grafana.NewStatPanel(&grafana.StatPanelOptions{
+    PanelOptions: &grafana.PanelOptions{
+        Datasource:  "Prometheus",
+        Title:       "Uptime",
+        Description: "instance uptime",
+        Span:        12,
+        Height:      4,
+        Decimals:    2,
+        Unit:        "s",
+        Query: []grafana.Query{
+            {
+                Expr:   `uptime_seconds`,
+                Legend: `{{ pod }}`,
+            },
+        },
+    },
+    ColorMode:   common.BigValueColorModeNone,
+    TextMode:    common.BigValueTextModeValueAndName,
+    Orientation: common.VizOrientationHorizontal,
+}))
+
+db, err := builder.Build()
+if err != nil {
+    return nil, err
+}
+json, err := db.GenerateJSON()
+if err != nil {
+    return nil, err
+}
+fmt.Println(string(json))
+```
```

Comment:
maybe consider making this an expandable section?
(and a real complete file w/ `func main`?)
<details><summary>main.go</summary>

```go
package main

import "github.com/smartcontractkit/chainlink-common/observability-lib/grafana"

func main() {
	builder := grafana.NewBuilder(&grafana.BuilderOptions{
	    Name:       "Dashboard Name",
	    Tags:       []string{"tags1", "tags2"},
	    Refresh:    "30s",
	    TimeFrom:   "now-30m",
	    TimeTo:     "now",
	})
	
	builder.AddVars(grafana.NewQueryVariable(&grafana.QueryVariableOptions{
	    VariableOption: &grafana.VariableOption{
	        Label: "Environment",
	        Name:  "env",
	    },
	    Datasource: "Prometheus",
	    Query:      `label_values(up, env)`,
	}))
	
	builder.AddRow("Summary")
	
	builder.AddPanel(grafana.NewStatPanel(&grafana.StatPanelOptions{
	    PanelOptions: &grafana.PanelOptions{
	        Datasource:  "Prometheus",
	        Title:       "Uptime",
	        Description: "instance uptime",
	        Span:        12,
	        Height:      4,
	        Decimals:    2,
	        Unit:        "s",
	        Query: []grafana.Query{
	            {
	                Expr:   `uptime_seconds`,
	                Legend: `{{ pod }}`,
	            },
	        },
	    },
	    ColorMode:   common.BigValueColorModeNone,
	    TextMode:    common.BigValueTextModeValueAndName,
	    Orientation: common.VizOrientationHorizontal,
	}))
	
	db, err := builder.Build()
	if err != nil {
	    return nil, err
	}
	json, err := db.GenerateJSON()
	if err != nil {
	    return nil, err
	}
	fmt.Println(string(json))
}
```
</details>

Reply:
yes very good idea 👍